src/registry/radio/helpers.ts

```diff
@@ -9,3 +10,27 @@ export const checkIsManualOptions = (
 ): component is RadioComponentSchema & {values: Option[] | undefined} => {
   return component.openForms.dataSrc === 'manual';
 };
+
+// A type guard is needed because TS cannot figure out it's a discriminated union
+// when the discriminator is nested.
+// See https://github.com/microsoft/TypeScript/issues/18758
+export const checkIsReferentielijstenOptions = (
+  component: RadioComponentSchema
+): component is RadioComponentSchema & {
+  data: {values: Option[] | undefined};
```

Comment:
for the builder, this means that the `values` key will be empty or missing in the radio component schema. For the renderer, it will be populated, but this is the builder here :) Also, the `values` key is directly included in radio, only `select` nests this under the `data` intermediate key.

```suggestion
```